modules/@apostrophecms/translation/index.js

```diff
@@ -13,9 +13,11 @@ module.exports = {
     return {
       '@apostrophecms/doc-type:beforeLocalize': {
         // Translate the document using the first available provider.
-        async translate(req, doc, source, target, options) {
+        async translate(req, doc, {
+          source, target, docType, existing
+        }) {
           const targets = req.body.aposTranslateTargets || [];
-          const translationProvider = self.apos.launder.string(req.body.aposTranslateProvider);
+          const providerName = self.apos.launder.string(req.body.aposTranslateProvider);
```

Comment:
The UI should add `query` parameters and we should check them here with `req.query.XXX`. We shouldn't modify the body in any way as this is the page data.